x/mongo/driver/operation/hello.go

```diff
@@ -305,17 +337,39 @@ func appendClientEnv(dst []byte, omitNonName, omitDoc bool) ([]byte, error) {
 		return bsoncore.AppendInt32Element(dst, "timeout_sec", timeoutInt32)
 	}
 
-	if !omitNonName {
-		switch name {
-		case envNameAWSLambda:
-			dst = addMem(envVarAWSLambdaFunctionMemorySize)
-			dst = addRegion(envVarAWSRegion)
-		case envNameGCPFunc:
-			dst = addMem(envVarFunctionMemoryMB)
-			dst = addRegion(envVarFunctionRegion)
-			dst = addTimeout(envVarFunctionTimeoutSec)
-		case envNameVercel:
-			dst = addRegion(envVarVercelRegion)
+	var idx int32
+	idx, dst = bsoncore.AppendDocumentElementStart(dst, "env")
+
+	if name != "" {
```

Comment:
Instead of nesting these conditions, suggest making them separate blocks for readability:

```go
	var idx int32
	idx, dst = bsoncore.AppendDocumentElementStart(dst, "env")

	if name != "" {
		dst = bsoncore.AppendStringElement(dst, "name", name)
	}

	if !omitNonName {
		switch name {
		case envNameAWSLambda:
			dst = addMem(envVarAWSLambdaFunctionMemorySize)
			dst = addRegion(envVarAWSRegion)
		case envNameGCPFunc:
			dst = addMem(envVarFunctionMemoryMB)
			dst = addRegion(envVarFunctionRegion)
			dst = addTimeout(envVarFunctionTimeoutSec)
		case envNameVercel:
			dst = addRegion(envVarVercelRegion)
		}
	}
```